ios/tournenizer/tournenizer/Services/TeamRequestService.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  TeamRequestService.swift
+//  tournenizer
+//
+//  Created by Ankush Rayabhari on 3/9/18.
+//  Copyright © 2018 Ankush Rayabhari. All rights reserved.
+//
+
+import Foundation;
+
+class TeamRequestService : Service {
+    static let shared = TeamRequestService();
+
+    func getRequestsForCurrentUser(cb: @escaping ((String?, [TeamRequest]?, [Tournament]?, [User]?, [Team]?) -> Void)) {
+        makeRequest(url: Constants.route.user.requests, type: .GET, body: Data(base64Encoded: "")) { (error: String?, data: Data?) in
+            if(error != nil) {
+                return cb(error, nil, nil, nil, nil);
+            }
+
+            let teamRequests: [TeamRequest]? = self.decode(data!);
+            if(teamRequests == nil) {
+                return cb(Constants.error.genericError, nil, nil, nil, nil);
+            }
+
+            var users = [User?](repeating: nil, count: teamRequests!.count);
+            var teams = [Team?](repeating: nil, count: teamRequests!.count);
+            var tournaments = [Tournament?](repeating: nil, count: teamRequests!.count);
+
+            let group = DispatchGroup();
+            var errorOccurred = AtomicBoolean();
```

Comment:
Why does this boolean have to be atomic?

Reply:
URLSession will always make the completion handler run on a background thread. So each of the TeamService, TournamentService, UserService could access it from different threads.